### ISSUE TITLE
OAuth2 not working

### ISSUE THREAD

**Issue by oculos** [roundcube/roundcubemail#9221]
Hi,

I configured Roundcube to work with OAuth2 against keycloak. Dovecot wasn't working, but it finally worked.

But then something weird happened:

- users who had 2FA enabled on Roundcube via the 2fa plugin got authenticated and were asked to provide the 2FA token. I could see on dovecot logs that the user was indeed authenticated there with Xoauth2. However, after providing the 2FA token, I get an error sayin "Session is invalid or expired".
- users without 2FA simply get a "OAuth2 failed. Try again later" - here, no log whatsoever, not in dovecot, not in keycloak, nowhere.

I wonder if there's a plugin conflict or something. I deactivated the 2FA plugin, but it didn't help.

I am on Roundcube 1.6.4, and these are the plugins I have installed:

Installed plugins
Plugin	Version	License	Source
acl	1.8	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
additional_message_headers	1.2.1	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
archive	3.5	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
emoticons	3.0	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
enigma	0.9	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
filesystem_attachments	1.0	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
identicon	0.1	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
identity_select	1.1	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
jqueryui	1.13.2	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
managesieve	9.4	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
markasjunk	2.0	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
newmail_notifier	0.8	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
sauserprefs	1.20	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	[Download](https://api.github.com/repos/johndoh/roundcube-sauserprefs/zipball/132f2442ee8531e53f892ccb941e3416a9f0e97f)
subscriptions_option	1.4	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
userinfo	1.2	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
vcard_attachments	4.2	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)	
zipdownload	3.4	[GPL-3.0+](http://www.gnu.org/licenses/gpl.html)

**Comment by alecpl** [comment]
Which 2fa plugin exactly? I'd say it will be up to the plugin to be compatible with a core feature.

**Comment by oculos** [comment]
> On Nov 21, 2023, at 07:49, Aleksander Machniak ***@***.***> wrote:
> 
> 
> Which 2fa plugin exactly? I'd say it will be up to the plugin to be compatible with a core feature.
> 

This one: 
alexandregz/twofactor_gauthenticator": "dev-master»

I mean, it was weird, but with this plugin installed, it worked halfway and I saw authentication on ldap. Without it, no IMAP authentication happens via oauth2 - just via username/password.

Best,
Francis

**Comment by alecpl** [comment]
I don't think we can/should fix this in the core. So, we can't help you. Ask the plugin author.

**Comment by oculos** [comment]
@alecpl I think you are misunderstanding the issue.

The issue is _not_ plugin related. The issue is about OAUTH2 authentication not working. My mention of the plugins was just to point out that it seemed to work halfway when a plugin was installed, and it doesn't work at all when the plugin is removed. 

Bottom line: oauth2 doesn't work for me - on the issue description I point out which error messages I get and which behavior I get.

**Comment by oculos** [comment]
never mind. it works now.
